leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java

```diff
@@ -139,6 +140,20 @@
                     throw new CodecException("One resource should be present in the payload [path:%s]", requestPath);
 
                 node = resourcesMap.values().iterator().next();
+            } else if (nodeClass == LwM2mResourceInstance.class) {
+                // validate we have resources for only 1 instance
+                if (jsonEntryByInstanceId.size() > 1)
+                    throw new CodecException("Only one instance expected in the payload [path:%s]", requestPath);
+
+                // Extract resources
+                Map<Integer, LwM2mResource> resourcesMap = extractLwM2mResources(
+                        jsonEntryByInstanceId.values().iterator().next(), baseName, model, requestPath);
+
+                // validate there is only 1 resource
+                if (resourcesMap.size() != 1)
+                    throw new CodecException("One resource should be present in the payload [path:%s]", requestPath);
+
+                node = resourcesMap.values().iterator().next().getInstance(requestPath.getResourceInstanceId());
```

Comment:
here we should raise an exception if :
- resource is not a multiple resource
- if the multiple resource is empty 
- if the multiple resource contains more than 1 elements.

Reply:
agree with two first points, as for the 3rd - multiple resource can contains multiple instances

Reply:
> as for the 3rd - multiple resource can contains multiple instances

Generally, you're right, but here this is the case where try to decode a `LwM2mResourceInstance`, so the payload must contains only one `LwM2mResourceInstance`. (like for resource above and instance can contain several resource but when you read a resource you expect only 1 resource in the payload)

Reply:
Done